src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx

```diff
@@ -59,8 +62,12 @@ export const HopTransactionStep = ({
       return
     }
 
-    await executeTrade()
-  }, [executeTrade, swapTxState])
+    // Only proceed to execute the trade if the promise is resolved, i.e the user has opened the
+    // Ledger app without cancelling
+    await checkLedgerAppOpen(tradeQuoteStep.sellAsset.chainId)
```

Comment:
The name of this seems a little off, as it only checks if the Ledger App is open _if_ it's necessary to do so.

Maybe something more to the effect of `checkLedgerAppOpenIfLedgerConnected` - or even have a separate control flow for when a Ledger wallet is connected (though this complicates the well encapsulated logic in the `useLedgerOpenApp` hook.

Reply:
I did choose to migrate to `checkLedgerAppOpenIfLedgerConnected` in [5ebddb0](https://github.com/shapeshift/web/pull/7386/commits/5ebddb09c9daa88ca597a49ff2603d4e1f2b2716)

I wanted to keep the logic encapsulated in the hook